Fix Github Issue #304

ricejasonf opened the pull request:
- Apparently std::array operator[] returns an lvalue reference when unpacking
    a temporary std::array. Using std::get<n> returns the
    proper reference type in the case of rvalues.

ldionne commented on test/issues/github_304.cpp:
```diff
@@ -0,0 +1,21 @@
+// Copyright Jason Rice 2016
```
Can we please rename this file  to `test/ext/std/array/issue_304.cpp`? This way, the dependency upon external adapters is explicit.

ricejasonf commented:
I added your file and renamed the other test. (I could remove it if you think it is redundant.)

Sorry about referencing the issue in the commit message. I forgot that it did that. :sweat_smile:

ldionne commented:
Thanks a lot! I'll merge this as soon as the CIs are done.

ldionne commented:
IDK why this was not closed automatically when I merged, but anyway it's merged as ea571a11fc5d3dc6fd2f73a0b1e7e9c0be2fc614.
